chalice/app.py

```diff
@@ -477,14 +482,16 @@ def _add_route(self, path, view_func, **kwargs):
         if kwargs:
             raise TypeError('TypeError: route() got unexpected keyword '
                             'arguments: %s' % ', '.join(list(kwargs)))
-        if path in self.routes:
-            raise ValueError(
-                "Duplicate route detected: '%s'\n"
-                "URL paths must be unique." % path)
-        entry = RouteEntry(view_func, name, path, methods,
-                           authorizer_name, api_key_required,
-                           content_types, cors, authorizer)
-        self.routes[path] = entry
+        for method in methods:
+            if method in self.routes[path]:
+                raise ValueError(
+                    "Duplicate method: '%s' detected for route: '%s'\n"
+                    "A specific method may only be specified once for "
+                    "a particular path." % (method, path))
```

Comment:
Sure. I should be able to have access to both view function names. So it should be feasible.